pkgs/tools/cd-dvd/ventoy-bin/default.nix

```diff
@@ -0,0 +1,107 @@
+{
+  lib, stdenv, fetchurl, fetchpatch,
+  autoPatchelfHook, makeWrapper,
+  hexdump, exfat, dosfstools, xz, util-linux, bash, parted
+}:
+
+stdenv.mkDerivation rec {
+  pname = "ventoy-bin";
+  version = "1.0.47";
+  meta = with lib; {
+    description =
+      "An open source tool to create bootable USB drive for ISO/WIM/IMG/VHD(x)/EFI files";
+    longDescription = ''
+      An open source tool to create bootable USB drive for
+      ISO/WIM/IMG/VHD(x)/EFI files.  With ventoy, you don't need to format the
+      disk over and over, you just need to copy the ISO/WIM/IMG/VHD(x)/EFI
+      files to the USB drive and boot them directly.  You can copy many files
+      at a time and ventoy will give you a boot menu to select them
+      (screenshot).  x86 Legacy BIOS, IA32 UEFI, x86_64 UEFI, ARM64 UEFI and
+      MIPS64EL UEFI are supported in the same way.  Most type of OS supported
+      (Windows/WinPE/Linux/Unix/VMware/Xen...).
+    '';
+    homepage = "https://www.ventoy.net";
+    license = licenses.gpl3;
+    platforms = [ "x86_64-linux" "i686-linux" "aarch64-linux" "mipsel-linux" ];
+    maintainers = with maintainers; [ k4leg ];
+  };
+
+  nativeBuildInputs = [ autoPatchelfHook makeWrapper ];
+  buildInputs = [ hexdump exfat dosfstools xz util-linux bash parted ];
+
+  src = fetchurl {
+    url = "https://github.com/ventoy/Ventoy/releases/download/v${version}/ventoy-${version}-linux.tar.gz";
+    sha256 = "1cc045f31779047384f767057458cc9da45a0015d1760bdbc43518fb6faad37d";
+  };
+  patches = [
+    (fetchpatch {
+      name = "sanitize.patch";
+      url =
+        let commit = "230f41afeb6939e215e7be6cedd2fb1da6c094a4";
+        in "https://aur.archlinux.org/cgit/aur.git/plain/sanitize.patch?h=ventoy-bin&id=${commit}";
+      sha256 = "d250816de0903a5fc5364ee25914a06b1b1595bcfc72dac84ad016e1d30727c4";
+    })
+    ./fix-for-read-only-file-system.patch
+    ./fix-path-to-log.patch
+  ];
+  patchPhase = ''
+    runHook prePatch
+
+    for patch in ''${patches[@]}; do
+        patch -p0 < "$patch"
+    done
+    find -type f -name \*.sh -exec chmod a+x {} \;  # Fix permissions.
+
+    runHook postPatch
+  '';
+  installPhase = ''
```

Comment:
```suggestion
    # Fix permissions
    find -type f -name \*.sh -exec chmod a+x {} \;
  '';

  installPhase = ''
```

Reply:
I don't know why but without this part 1-st patch (`sanitize.patch`) doesn't apply.
```bash
for patch in ''${patches[@]}; do
    patch -p0 < "$patch"
done
```

Reply:
Try the stripLen attribute for fetchpatch. Make sure to invalidate the hash to update it.